scripts/scene/RoomBuilder.gd

```diff
@@ -225,7 +225,7 @@ func generate_unique_room_id() -> int:
 	return unique_id
 
 func check_room_id_exists(room_id: int) -> bool:
-	return room_id in rooms
+	return rooms.any(func(room: Room): return room.id == room_id)
```

Comment:
previous function was generating errors

Reply:
Thanks. The old function doesn't make sense anymore. `rooms` should be an array of Room objects. I think this is a hangover from a simpler setup when we were just storing the ids, and I forgot to update it.